Fixed link to strict equality operator

simonvarey opened the pull request:
<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->
#### Summary
 I fixed the internal link to the strict equality operator page.

#### Motivation
The internal link to the strict equality operator page was broken.

#### Metadata
<!-- ✅ Check a box if applicable, like this: [x]

This PR…
-->
- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

<!-- 👷‍♀️ After submitting, review the results of the "Checks" tab! -->

simonvarey committed:
Fixed link to strict equality operator.

The internal link to the strict equality operator page was broken. I fixed the internal link to the strict equality operator page.

wbamberg reviewed (commented):
I wonder what the point is of including an example that isn't a good example.

Perhaps someone in the Wiki days included that example as a good example, then someone else added the bit about it not being a good example?

I think we should just delete this example completely. We already have a warning further up that `split` does not respect surrogates.

simonvarey committed:
Deleted Reversing Example

Removed example of using `split` to reverse a `String`, because it is an example of what *not* to do

simonvarey commented:
@wbamberg I've now removed the whole reversing strings example

wbamberg reviewed (approved):
👍 thank you @simonvarey !
